ugsync/src/main/java/org/apache/ranger/unixusersync/process/PolicyMgrUserGroupBuilder.java

```diff
@@ -801,7 +811,55 @@ private void delXUserGroupInfo(XUserInfo aUserInfo, XGroupInfo aGroupInfo) {
 		}
 
 	}
-	
+
+	private MUserInfo addMUser(String aUserName, List<String> groups) {
+		MUserInfo ret = null;
+		MUserInfo userInfo = new MUserInfo();
+
+		userInfo.setLoginId(aUserName);
+		userInfo.setFirstName(aUserName);
+		userInfo.setLastName(aUserName);
+
+		if (isUserInAdminGroup (aUserName, groups)) {
+			String[] userRoleList = { "ROLE_SYS_ADMIN" };
+			userInfo.setUserRoleList(userRoleList);
+		}
+
+		if (authenticationType != null && AUTH_KERBEROS.equalsIgnoreCase(authenticationType) && SecureClientLogin.isKerberosCredentialExists(principal, keytab)) {
+			try {
+				Subject sub = SecureClientLogin.loginUserFromKeytab(principal, keytab, nameRules);
+				final MUserInfo result = ret;
+				final MUserInfo userInfoFinal = userInfo;
+				ret = Subject.doAs(sub, new PrivilegedAction<MUserInfo>() {
+					@Override
+					public MUserInfo run() {
+						try {
+							return getMUser(userInfoFinal, result);
```

Comment:
please make sure this code adds a log entry to a log file use to audit each time user is added...

Reply:
I changed the debug log inside the getMUser function to an info log.